src/pages/resultsView/plots/PlotsTabUtils.tsx

```diff
@@ -130,24 +129,42 @@ export interface IScatterPlotSampleData {
     profiledMutations?:boolean;
     mutations: AnnotatedMutation[];
     copyNumberAlterations: AnnotatedNumericGeneMolecularData[];
+    xThresholdType?: '>'|'<'|undefined; // threshold symbol for x coordinate in scatter plot
```

Comment:
please remove xT and yT, move them into a new interface IScatterPlotSampleData that extends IPlotSampleData

Reply:
Ok, I have made two attempts in this direction and both were unsuccessful. The reason for this is that I cannot reconcile 1) specific data types accepted by the different plots and 2) the type requirements of many functions in PlotsTabUtils.

I think the problem lies in the fact that in the current state `sample data` is the only property on which data points are styled in plots. As such all styling related functions in PlotsTabUtils are typed for the `IPlotSampleData` interface. With the introduction on a value-dependent qualifier such as the thresholdType symbol, the description of `IPlotSampleData` as a stand-in for what in effect is a `IPlotStylingData` interface is not longer sufficient.

I am afraid that splitting up of `IPlotSampleData` into different types for scatterplot (two-dimensional) and boxscatter/water (one-dimensional) as per your suggestion is something I tried two times, but failed. As stated in the paragraph above the reason for this is that `IPlotSampleData` is treated as a `IPlotStylingData` interface. The bad naming of this interface suggests that sample attributes are incorrectly mixed with value attributes in my PR. These attributes seem apples and pears, but are similar in the fact that these are attributes used to provide styling to datapoints. For this reason I added this comment:
```
// this interface contains all attributes needed to provide correct styling of 
// graph elements (points, bars, ...)
```

I my assessment correct so-far?

What do you think of this proposition:
```
export interface IPlotSampleData {}

export interface ITwoDimentsionPlotStyling {
    xThresholdType?: '>'|'<'|undefined; // threshold symbol for x coordinate in scatter plot
    yThresholdType?: '>'|'<'|undefined; // threshold symbol for y coordinate in scatter plot
}

export interface IOneDimensionPlotStyling {
    thresholdType?: '>'|'<'|undefined; // threshold symbol for x coordinate in scatter plot
}

export interface IPlotStylingData extends IPlotSampleData, IOneDimensionPlotStyling, ITwoDimentsionPlotStyling {}
```
After this rework, we can remap all styling related function to the new `IPlotStylingData` interface. Does this rework make sense?

Reply:
Sorry but I'm confused. Can we talk on the phone about this? Can you speak for a half hour on Monday at 10 AM my time?

Reply:
Note: after discussion with @adamabeshouse we decided to refactor the interface and add  'thresholdType` type checking in the function in PlotsTabUtils that accept data points that conform to the less specific IPlotsTabData interface.